tests/integration/pause.tcl

```diff
@@ -0,0 +1,64 @@
+source tests/support/benchmark.tcl
+
+start_server {tags {"external:skip"}} {
+  set master_host [srv 0 host]
+  set master_port [srv 0 port]
+
+  test {Test client-pause-write-during-oom} {
+    # Scenario:
+    # When client-pause-write-during-oom is configured, if OOM and available pending 
+    # lazy-free jobs in background, then (clients are paused and) eviction is 
+    # suspended. The test will:
+    # - Fill up DB with many small items and configures tight maxmemory
+    # - FLUSHALL ASYNC  (slow background operation by lazyfree thread)    
+    # - Fill up DB with many big items (quick operation by main thread)
+    #  
+    # It is expected that filling up DB for the second time will be quicker than the
+    # FLUSHALL ASYNC release and will cause to OOM, and if client-pause-write-during-oom
+    # is configured then there won't be any eviction or only few 
+
+    set FEATURE_DISABLED no
+    set FEATURE_ENABLED yes
+
+    # Prepare cmd2 in advance in order to run as fast as possible after flushall
+    set cmd1 [redisbenchmark $master_host $master_port " -t set -n 990000 -r 100000000 -d 1 -P 150"]
+    set cmd2 [redisbenchmark $master_host $master_port " -t set -n 2200 -r 100000000 -d 20000 -P 20"]
+
+    foreach configVal [list $FEATURE_DISABLED $FEATURE_ENABLED] {
+        # Configure maxmemory to be tightly upper-bound to DB size, (~80.4MB)
+        r config set maxmemory 84000000
+        r config set client-pause-write-during-oom $configVal
+        r config set maxmemory-policy allkeys-random
+
+        # Restart && and fill up DB, with many small items up-to used_mem of 77MB
+        common_bench_setup $cmd1
+
+        # reset evicted_keys counter
+        r config resetstat
+        
+        # FLUSHALL expected to run slow in background due to small items.
+        r flushall async
+
+        # Fill up DB with many big items up-to used_mem  of ~44MB. Expected to run fast
+        exec {*}$cmd2
+
+        set evicted($configVal) [s evicted_keys]
+    }
+
+    # In an attempt to avoid from flaky test, we shall apply the following logic:
+    # If: there are only few evictions when feature is disabled 
+    # then: expected to see no eviction at all when feature is enabled
+    # else: expected at least four times more evictions when feature is disabled  
+    if { $evicted($FEATURE_DISABLED) < 12 } {
+        assert_equal $evicted($FEATURE_ENABLED) 0
+    } else {    
+        assert_lessthan [expr 4 * $evicted($FEATURE_ENABLED)] $evicted($FEATURE_DISABLED)
+    }
+    
+    # Verify that when feature is disabled the test counted evictions
+    # If test will become flaky, as some machine\executions might run FLUSHALL releases
+    # faster than DB reconstructions, then consider comment this line
+    assert_morethan $evicted($FEATURE_DISABLED) 0 
```

Comment:
i'm paranoid about these tests being flaky.
maybe we can write a specific test using two clients and one command in each in some way?
e.g.using (`KILL -STOP`) and some other tricks..

Reply:
I let it run on two platforms for 1000 iterations. 
Let's give it a try and see if it holds.

Reply:
i'm still afraid. when that test runs in parallel to other tests, it could happen that the one that was supposed to produce a lot of evictions didn't produce many, and the other one wasn't affected, but still had a race and produced a few, and the factor is less than 4x.

Reply:
simplified test. Please review again test.